packages/orchestration/test/mocks/chain-info.js

```diff
@@ -0,0 +1,304 @@
+/**
+ * Generated using gRPC queries to local simulated chains
+ * See {@link https://github.com/Agoric/agoric-sdk/blob/43e0249502171e82460be7cf4bad8baf5db5aafd/packages/cosmic-proto/test/snapshots/test-chain-info.js.md | snapshots} and
+ * their {@link https://github.com/Agoric/agoric-sdk/blob/43e0249502171e82460be7cf4bad8baf5db5aafd/packages/cosmic-proto/test/snapshots/test-chain-info.js.md | source test}.
+ * // TODO check in some portion of above code
+ */
+export const mockChainInfo = {
+  agoric: {
+    icaParams: {
+      allowMessages: [
+        '/agoric.*',
+        '/ibc.*',
+        '/cosmos.[a-rt-z]*',
+        '/cosmos.s[a-su-z]*',
+      ],
+      hostEnabled: true,
+    },
+    icqParams: {
+      hostEnabled: false,
+    },
+    chainId: 'agoriclocal',
+    bondDenom: 'ubld',
+    denoms: [
+      {
+        denom: 'ubld',
+        native: true,
+      },
+      {
+        denom: 'uist',
+        native: true,
+      },
+      {
+        baseDenom: 'uion',
+        denom:
+          'ibc/F7E92EE59B5428793F3EF5C1A4CB2494F61A9D0C9A69469D02390714A1372E16',
+        path: 'transfer/channel-0',
+      },
+      {
+        baseDenom: 'uosmo',
+        denom:
+          'ibc/ED07A3391A112B175915CD8FAF43A2DA8E4790EDE12566649D0C2F97716B8518',
+        path: 'transfer/channel-0',
+      },
+      {
+        baseDenom: 'uatom',
+        denom:
+          'ibc/C4CFF46FD6DE35CA4CF4CE031E643C8FDC9BA4B99AE598E9B0ED98FE3A2319F9',
+        path: 'transfer/channel-1',
+      },
+    ],
+    connections: [
+      {
+        chainId: 'osmosis-test',
+        connectionInfo: {
+          clientId: '07-tendermint-2',
+          counterparty: {
+            clientId: '07-tendermint-2',
+            connectionId: 'connection-1',
+          },
+          id: 'connection-0',
+          state: 'STATE_OPEN',
+        },
+        lastUpdated: {
+          revisionHeight: 831n,
```

Comment:
Pretty frequently, I believe with every block (of the chain with the shorter avg block time).

I recognize this likely makes it a poor candidate to put into state, given our anticipated update frequency/governance, but some messages seem like they rely on it for preventing an unbound timeout:

1. [MsgTransfer](https://buf.build/cosmos/ibc/docs/1297551504e3c2915a97459d83522b5f424c8a81:ibc.applications.transfer.v1#ibc.applications.transfer.v1.MsgTransfer) has `timeoutHeight` and `timeoutTimestamp` fields (_the generated TS types say both are required, but i think it's actually one or the other_)
2. [RequestQuery](https://buf.build/tendermint/tendermint/docs/main:tendermint.abci#tendermint.abci.RequestQuery) has a `height` field used for timing out a packet
3. [TxBody](https://buf.build/cosmos/cosmos-sdk/docs/main:cosmos.tx.v1beta1#cosmos.tx.v1beta1.TxBody) has a `timeoutHeight` field used for timing out a packet


More on `Height`:
```ts
/**
 * Height is a monotonically increasing data type
 * that can be compared against another Height for the purposes of updating and
 * freezing clients
 *
 * Normally the RevisionHeight is incremented at each height while keeping
 * RevisionNumber the same. However some consensus algorithms may choose to
 * reset the height in certain conditions e.g. hard forks, state-machine
 * breaking changes In these cases, the RevisionNumber is incremented so that
 * height continues to be monitonically increasing even as the RevisionHeight
 * gets reset
 */
export interface Height {
    /** the revision that the client is currently on */
    revisionNumber: bigint;
    /** the height within the given revision */
    revisionHeight: bigint;
}
```

For `MsgTransfer`, it seems that we should use `timeoutTimestamp` (absolute) to set the timeout, leaning on `TimerService`. 

For `RequestQuery` and `TxBody`, maybe the correct abstraction for now is to ensure `timeoutHeight/height` is always an `(invitation|offer)Arg`, and rely on clients for this information.

A clever approach might be to approximate timeout height based on a target block duration (of the chain with the shorter avg block times), using using a historical revisionHeight and the timestamp is was saved at. This is likely too clever, and doesn't account for things like chain halting upgrades or target block time drift.


P.S. created #9324 to capture this work, as `TxBody`'s `revisionHeight` and `RequestQuery`'s `height` default to `0n` on `master`